frontend/app/helpers/spree/structured_data_helper.rb

```diff
@@ -0,0 +1,56 @@
+module Spree
+  module StructuredDataHelper
+    def products_structured_data(products)
+      content_tag :script, type: 'application/ld+json' do
+        raw(
+          products.map do |product|
+            structured_product_hash(product)
+          end.to_json
+        )
+      end
+    end
+
+    private
+
+    def structured_product_hash(product)
+      {
+        '@context': 'https://schema.org/',
+        '@type': 'Product',
+        '@id': "#{engine_routes_helper.root_url(host: current_store_host)}product_#{product.id}",
```

Comment:
@mbajur in a helper you can use helper routes directly without passing the host

Reply:
i'm using current store host in here (i was not aware of any possible drawbacks of not doing so so i followed @piotrleniec-spark route on that).

So, to be sure - you want me to drop usage of current store host and just use the detected request host ?